### PR TITLE
Update Next Steps with more up-to-date information

### PR DESCRIPTION
**Pull request by Menardi**
### Platforms affected
Docs

### What does this PR do?
Recently I was reading the `Next Steps` page on the docs, and noticed it was very outdated. For example, it references things like iOS 6 and Cordova 3 as if they are brand new. I have updated the content to remove outdated information, and bring other stuff up to date.

I have also removed links to jQuery Mobile, as it is no longer really being maintained. I personally don't think we should be encouraging people in 2018 to make jQuery Mobile apps! I have also added links to Vue, and Onsen UI (full disclosure, I am a maintainer of Onsen UI).

### What testing has been done on this change?
Previewed locally in browser.

### Checklist
- [ ] Commit message follows the format: "GH-3232: (android) Fix bug with resolving file paths", where GH-xxxx is the GitHub issue ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: www/docs/en/dev/guide/next/index.md [hunk 1]
```diff
@@ -23,85 +23,62 @@ description: A look at topics that new Cordova developers will encounter.
...
+**Click versus Touch** - Many devices impose a 300ms delay on click events in order to distinguish between a click and double click event. This can have the effect of making your app feel slow and unresponsive. Avoiding this delay is one of the most important ways of improving your app's perceived performance. Most Android versions no longer impose this delay, but iOS by default still does. For iOS, using [WkWebView](https://github.com/apache/cordova-plugin-wkwebview-engine) instead of the default UIWebView gets rid of this delay. If you need to support older devices (Android 4.4 or iOS 8), consider a polyfill such as [FastClick](https://github.com/ftlabs/fastclick), or using `touchstart` and `touchend` instead of `click`.
```
Comment (dpogue) [inline www/docs/en/dev/guide/next/index.md]: Maybe worth mentioning that this depends on the value in a viewport tag (in fact, maybe the viewport meta tag should be its own heading in here)

File: www/docs/en/dev/guide/next/index.md [hunk 2]
```diff
@@ -23,85 +23,62 @@ description: A look at topics that new Cordova developers will encounter.
...
-["Is This Thing On?"](http://blogs.telerik.com/appbuilder/posts/13-04-23/is-this-thing-on-%28part-1%29)
+Note that the online and offline events, as well as the Network Connection API, is not perfect. You may need to rely on using an XHR request to see if the device is truly offline or online. At the end of the day, be sure add some form of support for network issues - in fact, the Apple store (and probably other stores) will reject apps that don't properly handle offline/online states. For more discussion on this topic, see
+["Is This Thing On?"](https://blogs.telerik.com/appbuilder/posts/13-04-23/is-this-thing-on-%28part-1%29)
 
 # Handling Upgrades
 
 ## Upgrading Cordova Projects
 
-If your existing project was created using Cordova 3.x, you can upgrade the project by issuing the following:
+You can upgrade the project by running the following:
```
Comment (dpogue) [inline www/docs/en/dev/guide/next/index.md]: I don't know that this command actually works... I seem to recall that it either renames the platform directory and installs a newer version beside it, or that it tries to actually upgrade the platforms files and fails horribly.

File: www/docs/en/dev/guide/next/index.md [hunk 3]
```diff
@@ -114,80 +91,77 @@ cordova plugin add "some-plugin"
...
+It is, of course, impossible to test on every possible device on the market. For this reason, it's wise to recruit many testers who have different devices. Although they won't catch every problem, chances are good that they will discover quirks and issues that you would never find alone.
 
-Tip: It is possible on Android Nexus devices to easily flash different versions of Android onto the device. This simple process will allow you to easily test your application on different levels of Android with a single device, without voiding your warranty or requiring you to “jailbreak” or “root” your device. Refer to the [Google Android factory images and instructions](https://developers.google.com/android/nexus/images#instructions).
+Tip: It is possible on Google Pixel devices to easily flash different versions of Android onto the device. This simple process will allow you to easily test your application on different levels of Android with a single device, without voiding your warranty or requiring you to “jailbreak” or “root” your device. Refer to the [Google Android factory images and instructions](https://developers.google.com/android/images).
```
Comment (dpogue) [inline www/docs/en/dev/guide/next/index.md]: I would remove this bit entirely.

The Android emulator has improved ***significantly*** in the past few years, to the point I suspect people are better off using that for basic testing as opposed to flashing different versions onto an expensive phone.

File: www/docs/en/dev/guide/next/index.md (hunk 1 above)
Comment (Menardi) [reply www/docs/en/dev/guide/next/index.md]: I forgot about the need for the `meta` tag, I'll add a bit about that.

**Comment by Menardi** [comment]
Thanks for the review, @dpogue. I've updated the PR to remove the things you mentioned, and also added info about needing the meta tag to prevent the 300ms tap delay.

**Pull request merged**